From foo@baz Tue Apr  9 12:12:43 2002
To: Greg KH <greg@kroah.com>
Date: 07 Feb 2005 16:20:50 -08:00
From: bjorn.helgaas@hp.com
Subject: [PATCH] PCI: NUMA-Q PCI config access arg validation

Fix NUMA-Q PCI config access bus validation.  "bus" indexes
into BUS2QUAD, which is mp_bus_id_to_node[MAX_MP_BUSSES].

This depends on the "pci_raw_ops should use unsigned args"
patch I posted earlier today (no functional dependency; it
just happens to be very close textually).

Signed-off-by: Bjorn Helgaas <bjorn.helgaas@hp.com>
Signed-off-by: Martin J. Bligh <mbligh@aracnet.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


diff -Nru a/arch/i386/pci/numa.c b/arch/i386/pci/numa.c
--- a/arch/i386/pci/numa.c	2005-03-02 23:05:08 -08:00
+++ b/arch/i386/pci/numa.c	2005-03-02 23:05:08 -08:00
@@ -18,7 +18,7 @@
 {
 	unsigned long flags;
 
-	if (!value || (bus > MAX_MP_BUSSES) || (devfn > 255) || (reg > 255))
+	if (!value || (bus >= MAX_MP_BUSSES) || (devfn > 255) || (reg > 255))
 		return -EINVAL;
 
 	spin_lock_irqsave(&pci_config_lock, flags);
@@ -46,7 +46,7 @@
 {
 	unsigned long flags;
 
-	if ((bus > MAX_MP_BUSSES) || (devfn > 255) || (reg > 255)) 
+	if ((bus >= MAX_MP_BUSSES) || (devfn > 255) || (reg > 255)) 
 		return -EINVAL;
 
 	spin_lock_irqsave(&pci_config_lock, flags);
